hanish/bot.py

```diff
@@ -0,0 +1,409 @@
+        """
+        # Parse the location command arguments
+        for match in self.commands['location'].finditer(msg['text']):
+            zipcode = match.group(1)
+            weather = self.weather(zipcode)
+
+            response = weather_currently(weather)
+            self.post(msg['channel'], response)
+
+    def handle_darksky_command(self, msg):
+        """
+        Have the chatbot respond to the darksky command.
+        """
+        response = "I have made {} Dark Sky API calls today.".format(
+            self.darksky.n_api_calls
+        )
+        self.post(msg['channel'], response)
+
+    def handle_unknown_command(self, msg):
+        """
+        Have the chatbot respond when an unknown command is sent.
+        """
+        response = (
+            "I'm happy to chat about the weather, "
+            " unfortunately I don't understand what you're asking."
+        )
+
+        self.post(msg['channel'], response)
+
+    def post(self, channel, response):
+        """
+        Helper function to post a message as the slackbot.
+        """
+        self.slack.api_call(
+            "chat.postMessage", channel=channel, text=response, as_user=True
+        )
+
+    def set_schedule(self):
+        """
+        Sets the schedule to do tasks on demand.
+        """
+        schedule.every().day.at("9:00").do(self.weather_notice)
+
+    def weather(self, zipcode=None):
+        """
+        Quick lookup of the current weather for a given zipcode. If no zipcode
+        is supplied, then will look up the weather for the default zipcode.
+
+        Parameters
+        ----------
+        zipcode: string, default None
+            Zip Code to lookup weather for or None for the default
+
+        Returns
+        -------
+        data: json
+            Prints a nicely idented summary of the weather
+
+        msg: string
+            Returns a string of the number of dark sky api calls made.
+        """
+        # Resolve the latitutde and longitude from the zipcode and query
+        # weather. NOTE: this will utilize cacheing on the API if available.
+        zipcode  = zipcode or self.zipcode
+        lat, lon = self.zipdb.lookup(zipcode)
+        forecast = self.darksky.forecast(lat, lon)
+
+        # Add additional information and return
+        forecast['zipcode'] = zipcode
+        forecast['api_calls'] = self.darksky.n_api_calls
+        return forecast
+
+    def weather_notice(self):
+        """
+        Detects if there is any material change in the weather and alerts the
+        channel. This method is scheduled to run every day at 9:00am
+        """
+        # Keep track if we should alert or not, and get the current weather.
+        alert   = False
+        weather = self.weather()
+
+        if self.yesterday:
+            # Fetch weather for yesterday and today
+            current = weather['hourly']['summary']
+            yesterday = self.yesterday['hourly']['summary']
```

Comment:
It seems like `self.yesterday` is only set by this block, so given that weather is always the configured zipcode, it should be ok.